pkg/daemonsetlock/daemonsetlock.go

```diff
@@ -137,6 +144,24 @@ func (dsl *DaemonSetLock) Release() error {
 	}
 }
 
+// GetDaemonSet returns the named DaemonSet resource from the DaemonSetLock's configured client
```

Comment:
@squaremo this implementation includes a generic `GetDaemonSet` method that is invoked in common in the three appropriate places above